checks/ad_replication

```diff
@@ -99,32 +101,39 @@ def check_ad_replication(item, params, info):
 
             if int(num_failures) > max_failures_warn:
                 status = 1
+                count_failures = count_failures + num_failures
+                count_failed_repl = count_failed_repl + 1
                 output += '(!) %s/%s replication of context %s reached ' \
                           ' the threshold of maximum failures (%s) (Last success: %s, ' \
-                          'Last failure: %s, Num failures: %s, Status: %s), ' % \
+                          'Last failure: %s, Num failures: %s, Status: %s)\n' % \
                          (source_site, source_dc, naming_context, max_failures_warn,
                           time_last_success_txt, time_last_failure_txt, num_failures, statusLastFailure)
 
             if int(num_failures) > max_failures_crit:
                 status = 2
+                count_failures = count_failures + num_failures
+                count_failed_repl = count_failed_repl + 1
                 output += '(!!) %s/%s replication of context %s reached ' \
                           ' the threshold of maximum failures (%s) (Last success: %s, ' \
-                          'Last failure: %s, Num failures: %s, Status: %s), ' % \
+                          'Last failure: %s, Num failures: %s, Status: %s)\n' % \
                          (source_site, source_dc, naming_context, max_failures_crit,
                           time_last_success_txt, time_last_failure_txt, num_failures, statusLastFailure)
 
             if time_last_failure is not None and time_last_success is not None \
                and time_last_failure > time_last_success:
                 status = 2
+                count_failures = count_failures + 1
+                count_failed_repl = count_failed_repl + 1
                 output += '(!!) %s/%s replication of context %s failed ' \
-                         '(Last success: %s, Last failure: %s, Num failures: %s, Status: %s), ' % \
+                         '(Last success: %s, Last failure: %s, Num failures: %s, Status: %s)\n' % \
                          (source_site, source_dc, naming_context,
                           time_last_success_txt, time_last_failure_txt, num_failures, statusLastFailure)
 
     if not foundLine:
         return (3, 'Replication information for %s not found' % item)
 
     if status != 0:
+        output = "%s replications had failures. %s total replication failures\n%s" % (count_failed_repl, count_failures, output)
```

Comment:
Please use a separate var for the long output, eg.:
long_output = []
long_output.append(TXT)
"\n".join(long_output)

Do not add the long output to the output in the same return-statement: The state marker would be added at the end (in the long output) but should be visible in the service details. In order to achieve that please use yield-statements:
if not foundLine:
  yield 3, 'Replication information not found'
  return

if status:
  yield status, 'TXT'
  if long_output:
    yield 0, '\n%s' % '\n'.join(long_output)
  return

yield 0, 'All replications are OK.'

Reply:
a hidden function, nice ;-)

I will implement this.